tsconfig.app.json

```diff
@@ -1,9 +1,11 @@
 {
-  "extends": "./tsconfig.base.json",
+  "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "./out-tsc/app",
-    "types": []
+    "types": [],
+    "strictNullChecks": true
```

Comment:
extending a strict config anyway, why change? Or the explicit excludes which shouldn't be needed (they're only needed due to the changed include). There's a number of changes like this - not problematic, but shouldn't be necessary. E.g. adding entry components where there weren't any before. Did some tool add these things?

Reply:
(not a blocker though - main issue is the failing dep install)

Reply:
Yeah, this might have been added by ng update scripts. I initially worked on it 3 months ago, so i don't recall. Let me try updating it.

Reply:
> (not a blocker though - the main issue is the failing dep install)

Yeah, the dep install will fail. The fix is in the follow-up PR. I will have to combine all three PRs into 1 before I can merge it with the main branch. Please approve if it looks good.

Reply:
go ahead and combine them and will do a final pass on the merge result. For other repos, is it possible to reverse the order or otherwise decouple them such that each once can be merged independently?

Reply:
Yeah, actually angular 14+ forces us to do the eslint migration. Similarly, had to migrate jest due to dependency issues.

Reply:
what about reversing the order though? Move to eslint on existing code, update jest, then angular 14

Reply:
Angular-eslint package's version will be dependent on the newer angular version.

Reply:
If not, then I will have to do double migration.

Reply:
I can try that if you want.

Reply:
If it's not double work, I think we should try it - it should be possible to use eslint with our current angular version. It's not the end of the world if we can't but I think it'll make reviewing it much quicker and thus easier for you with less time stuck in review for the larger repos.

Reply:
We can try that. Now that we have disabled all the new eslint rules, i don't think we will see many changes in ht-ui if we use the same rules set. for ht-ui, I will do the eslint first and then angular upgrades unless i hit a roadblock.